src/main/java/org/snomed/snowstorm/core/data/domain/Annotation.java

```diff
@@ -0,0 +1,133 @@
+		return annotationType != null ? annotationType.getPt() : null;
+	}
+
+	public void setAnnotationType(ConceptMini annotationType) {
+		this.annotationType = annotationType;
+	}
+
+	@Override
+	public String getAnnotationValue() {
+		return annotationValue;
+	}
+
+	public void setAnnotationValue(String annotationValue) {
+		this.annotationValue = annotationValue;
+	}
+
+	@Override
+	public String getAnnotationLanguage() {
+		return annotationLanguage;
+	}
+
+	public void setAnnotationLanguage(String annotationLanguage) {
+		this.annotationLanguage = annotationLanguage;
+	}
+
+	public Annotation fromRefsetMember(ReferenceSetMember fromMember) {
+		setAnnotationId(fromMember.getMemberId());
+		setMemberId(fromMember.getMemberId());
+		setRefsetId(fromMember.getRefsetId());
+		setModuleId(fromMember.getModuleId());
+		setActive(fromMember.isActive());
+		setEffectiveTimeI(fromMember.getEffectiveTimeI());
+		setReferencedComponentId(fromMember.getReferencedComponentId());
+		setReleased(fromMember.isReleased());
+		setAnnotationTypeId(fromMember.getAdditionalField(AnnotationFields.ANNOTATION_TYPE_ID));
+
+		String annotationValue = fromMember.getAdditionalField(AnnotationFields.ANNOTATION_VALUE);
+		setAnnotationValue(annotationValue);
+		if (StringUtils.isNotEmpty(annotationValue) && annotationValue.startsWith("@")) {
+			setAnnotationLanguage(annotationValue.substring(1, annotationValue.indexOf("\"")));
+		}
+		return this;
+	}
+
+	public ReferenceSetMember toRefsetMember() {
+		String refsetId = getRefsetId() != null ? getRefsetId() : Concepts.ANNOTATION_REFERENCE_SET;
+		String annotationId = getAnnotationId() != null ? getAnnotationId() : UUID.randomUUID().toString();
+		ReferenceSetMember member = new ReferenceSetMember(annotationId, getEffectiveTimeI(), isActive(), getModuleId(), refsetId, getReferencedComponentId());
+		member.setAdditionalField(AnnotationFields.ANNOTATION_TYPE_ID, getAnnotationTypeId());
+		member.setAdditionalField(AnnotationFields.ANNOTATION_VALUE, getAnnotationValue());
+		return member;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (!(o instanceof Annotation that)) return false;
+		if (!super.equals(o)) return false;
+		return Objects.equals(getReferencedComponentId(), that.getReferencedComponentId()) && Objects.equals(getAnnotationTypeId(), that.getAnnotationTypeId()) && Objects.equals(getAnnotationValue(), that.getAnnotationValue());
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(super.hashCode(), getReferencedComponentId(), getAnnotationTypeId(), getAnnotationValue());
+	}
+
+	@Override
+	public String toString() {
+		return "Annotation{" +
+				"annotationId='" + annotationId + '\'' +
+				", annotationTypeId='" + annotationTypeId + '\'' +
+				", annotationType=" + annotationType +
+				", annotationValue='" + annotationValue + '\'' +
+				", annotationLanguage='" + annotationLanguage + '\'' +
+				", active=" + active +
+				'}';
+	}
```

Comment:
@QuyenLy87 as well as the annotation specific properties, should we have the module, effective time etc in the toString method?